### PR TITLE
Drop Python 3.6 support and update some dependencies

### PR DESCRIPTION
**Pull request by jonasclaes**
This PR is similar to #1018.

- - -
This change is [Reviewable](https://reviewable.io/reviews/google/openhtf/1027)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -135,12 +135,12 @@ def run(self):
     'dataclasses;python_version<"3.7"',
```
Comment (lalten) [inline setup.py:135]: Should this also be removed?

Comment (jonasclaes) [reply setup.py:135]: Yes that seems only logical. Looks like I overlooked that one.

**Comment by jonasclaes** [comment]
Looks like I need to fix the mfg_inspector OAuth client.

**Comment by jonasclaes** [comment]
Can we try running the CI again? It should work now.

**Review by lalten — APPROVED** [review]

**Comment by jonasclaes** [comment]
LGTM!

**Comment by dbhatman** [comment]
https://github.com/google/openhtf/pull/1028 indicates your change would break the web gui due to tornado api changes.

**Comment by jonasclaes** [comment]
I have just reverted the upgrade of Tornado. Can we do another CI run?

**Comment by dbhatman** [comment]
I *think* the newest tornado version is required for python3.10 (https://github.com/tornadoweb/tornado/pull/2946). But sadly CI isn't catching it because of a lack of tests? We may want to wait to add py3.10 unfortunately :(

**Comment by jonasclaes** [comment]
Oh, hmm. Let's remove Python 3.10 for now then, I'll get started behind the scenes to convert this to be compatible with the new Tornado version.

**Review by cricdecyan — APPROVED** [review]

**Pull request merged**